### DISCUSSION TITLE
Support `DEFAULT` value / auto-`UUID` column in `CREATE` (similar to `SERIAL`)

### DISCUSSION THREAD

**Discussion by sapalli2989** [kuzudb/kuzu#3125] [Ideas]
It would be great, if we could declare within `CREATE`/DDL, that ids of a `UUID` column are to be auto-created via `gen_random_uuid()`. This feature would be analogue to `SERIAL` for integer sequences.

Currently Kuzu provides `gen_random_uuid()` for UUIDv4 auto-creation. But the need to invoke this function manually for new nodes  with `UUID` primary key is inconvenient and sometimes not easily possible - as seen with `MERGE` in https://github.com/kuzudb/kuzu/issues/3014. Actually I think proposal here is more useful than linked one.

PostgreSQL provides [`DEFAULT`](https://www.postgresql.org/docs/current/ddl-default.html) value keyword within `CREATE` in combination with equally named [`gen_random_uuid()`](https://www.postgresql.org/docs/current/ddl-default.html) for these cases (more examples in [1](https://stackoverflow.com/a/67945426), [2](https://dba.stackexchange.com/questions/122623/default-value-for-uuid-column-in-postgres)):
```sql
CREATE TABLE table_name (
    unique_id UUID DEFAULT gen_random_uuid(),
    ...
    PRIMARY KEY (unique_id)
);
```
or with directly prepended `PRIMARY KEY`:
```sql
CREATE TABLE table_name (
    unique_id UUID DEFAULT gen_random_uuid() PRIMARY KEY,
    ...
);
```
, requested for Kuzu in  https://github.com/kuzudb/kuzu/issues/2911.

I think this would be a nice addition to usability of `UUID`s.

**Comment by prrao87** [comment]
I think this is a really good idea, and I'm also wondering if it makes sense to have an alternative option to auto-create a unique integer primary key (something like UINT128, kind of what Neo4j does) instead of a UUID in cases where a simple serial integer works. From a user perspective, this is something that commonly is a source of annoyance: having to create a primary key manually for cases where the raw data doesn't have it.

But I'll leave it to @ray6080 and @andyfengHKU to discuss on any potential implementation challenges for this.

**Comment by sapalli2989** [comment]
See https://github.com/kuzudb/kuzu/issues/3238

**Comment by rjurney** [comment]
I want this. This is an obvious feature :)

**Reply by prrao87** [reply]
Hi @rjurney, this should work in the latest release:

```py
import kuzu

def main():
    db = kuzu.Database()
    conn = kuzu.Connection(db)

    # Create schema
    conn.execute("""
    CREATE NODE TABLE X (
        unique_id UUID DEFAULT gen_random_uuid() PRIMARY KEY,
        name STRING
    );
    """
    )
    conn.execute("MERGE (a:X {name: 'Alice'})")
    conn.execute("MERGE (a:X {name: 'Bob'})")

    # Check
    response = conn.execute(
        """
        MATCH (a:X)
        RETURN a.*
        """
    )
    for row in response:
        print(row)

if __name__ == "__main__":
    main()
```

```
[UUID('4bd87d6f-8d42-4905-ada1-7858d50f1621'), 'Alice']
[UUID('44eee4c0-3436-4382-b883-f2198c32efa0'), 'Bob']
```

**Reply by rjurney** [reply]
@prrao87 ahhh thanks!
